src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -790,6 +790,10 @@ DistributedTableCacheEntry(Oid distributedRelationId)
 	else
 	{
 		char *relationName = get_rel_name(distributedRelationId);
+
+		/* just in case */
+		Assert(relationName != NULL);
```

Comment:
the comment doesn't help much, maybe remove.

Also, should we do `ereport(ERROR, (errmsg("relation %s is not distributed", relationName ? relationName : "null")));`

for the edge cases that you've found like complex transactions with drop table

Reply:
converting the assertion to `ereport`